app/scripts/controllers/metametrics.ts

```diff
@@ -1143,29 +1304,29 @@ export default class MetaMetricsController {
           payload: {
             ...modifiedPayload,
             timestamp: modifiedPayload.timestamp.toString(),
-          },
+          } as SegmentEventPayload,
```

Comment:
It looks like the reason why this is needed is that this method allows the `event` property in `payload` to be missing, but the `state` does not.

I am not sure how best to handle this but perhaps we want to be more explicit here and add a `@ts-expect-error` that acts as a TODO and explains the problem?